jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -95,6 +98,44 @@ private static InsnNode simplifyInsn(MethodNode mth, InsnNode insn) {
 				}
 				break;
 
+			case CONSTRUCTOR:
+				ConstructorInsn constructor = (ConstructorInsn) insn;
+				if (constructor.getCallMth().getDeclClass().getType().equals(ArgType.STRING)) {
+					InsnArg arg = insn.getArg(0);
+					InsnNode node = arg.isInsnWrap()
+							? ((InsnWrapArg) arg).getWrapInsn()
+							: ((RegisterArg) arg).getParentInsn();
```

Comment:
`((RegisterArg) arg).getParentInsn()` is exactly `insn` and you don't need it, only wrapped insn